src/Installers/Windows/Wix.props

```diff
@@ -1,7 +1,13 @@
 <Project>
 
   <PropertyGroup>
-    <Version>$(AspNetCoreMajorVersion).$(AspNetCoreMinorVersion).$(AspNetCorePatchVersion).$(BuildNumber)</Version>
+    <!-- Ugly hack - Wix doesn't accept build numbers larger than 65535, and AzDo's BuildIds are larger than that.
```

Comment:
This hack kinda sucks, but I couldn't think of anything better - @joeloff @dougbu @JunTaoLuo any ideas?

Reply:
I think in TC the build number was literally just a counter. If ```BuildNumber``` is used for MSIs, then it doesn't really matter that much because Windows Installer only cares about the first 3 parts. For standalone bundles (EXEs), the 4 part version is considered when evaluating upgrades so having an increasing number is desired.

Are the AzDo numbers incremental as well? Unless it hits something large, e.g. 262140 and the modulus wraps, I guess this would be fine.

Reply:
Yeah, AzDO numbers are also incremental. We feed the version into `GuidInputs` as of https://github.com/dotnet/aspnetcore/pull/27079. Since the max for any part of the 4-part version, we'd run into possible issues as soon as any build number got past 65535 - it just might happen sooner in AzDO since builds are more frequent there than in TeamCity. But theoretically, the odds of a collision should be the same as the previous odds of an overflow

Reply:
And we'll always be incrementing the 3rd part of the 4-part version anyways (each patch)

Reply:
I think this is probably the best we can do for now but I'm a little concerned about the modulo logic since it can eventually silently cause issues. Instead of modulo maybe just use subtraction and error out if our build numbers are too large so we would be notified and figure something else out at that time? Though I don't know if that will ever happen, I feel it's safer since it'll fail loudly.

Reply:
I think the only risk is if we produce two versions that are exactly the same, which would happen if 2 builds of the same patch were exactly 65,535 AzDO builds apart (which seems unlikely). I believe 2.1.25.1 would still be considered "higher" than 2.1.24.65535 by WiX - @joeloff does that sound right? That is, it isn't strictly an issue if the `buildNumbers` aren't always increasing?

Reply:
This _would_ be an issue if we were producing previews of this branch, because there wouldn't be a guarantee that the versions would always be increasing (e.g. preview1's `BuildNumber` could be higher than preview2's, and they'd have the same patch number)